tests/test_location_pi.py

```diff
@@ -0,0 +1,508 @@
+            self.test_location_01._validate_perpetual_inventory_request(req)
+
+    def test05_process_pi_datetime_no_changes(self):
+        """
+        Both last check & last correct PI datetime are updated if
+        the PIOutcome is empty
+        """
+        outcome = MockPIOutcome(False)
+        old_dt_checked = self.test_location_01.u_date_last_checked
+        old_dt_success = self.test_location_01.u_date_last_checked_correct
+        time.sleep(1)
+
+        self.test_location_01._process_pi_datetime(outcome)
+
+        self.assertNotEqual(old_dt_checked,
+                            self.test_location_01.u_date_last_checked,
+                            "The checked datetime was not updated.")
+
+        self.assertNotEqual(old_dt_success,
+                            self.test_location_01.u_date_last_checked_correct,
+                            "The last correct datetime was not updated.")
+
+    def test06_process_pi_datetime_with_inventory_changes(self):
+        """
+        Only the last check datetime is updated if the PIOutcome
+        is not empty
+        """
+        outcome = MockPIOutcome(True)
+        old_dt_checked = self.test_location_01.u_date_last_checked
+        old_dt_success = self.test_location_01.u_date_last_checked_correct
+        time.sleep(1)
+
+        self.test_location_01._process_pi_datetime(outcome)
+
+        self.assertNotEqual(old_dt_checked,
+                            self.test_location_01.u_date_last_checked,
+                            "The checked datetime was not updated.")
+
+        self.assertEqual(old_dt_success,
+                         self.test_location_01.u_date_last_checked_correct,
+                         "The last correct datetime was wrongly updated.")
+
+    #
+    ## PI count moves
+    #
+
+    def _check_picking_loc01_to_loc02(self, picking, picking_idx):
+        self.assertEqual(picking.location_id.id, self.test_location_01.id,
+                         "Picking %d has wrong source location" % picking_idx)
+        self.assertEqual(picking.location_dest_id.id, self.test_location_02.id,
+                         "Picking %d has wrong dest location" % picking_idx)
+        self.assertTrue(picking.has_packages,
+                        "Picking %d does not have packages" % picking_idx)
+        self.assertEqual(len(picking.move_line_ids), 1,
+                         "Picking %d has not single move line" % picking_idx)
+
+    def test07_process_pi_count_moves_valid_request_with_package(self):
+        """ Successfully creates the expected picking """
+        self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                          package_id=self.package_one.id)
+        count_moves = [{"package_id": self.package_one.id,
+                        "location_id": self.test_location_01,
+                        "location_dest_id": self.test_location_02.id}]
+
+        pickings = self.test_location_01._process_pi_count_moves(count_moves)
```

Comment:
this probably needs to be test_location_02._process_pi_.... since we are moving the package from 01 to 02 because we are pi counting test02 and the package is in the wrong location.